### PR TITLE
libsForQt5: use makeScopeWithSplicing

### PR DESCRIPTION
**Pull request by Artturin**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by ghost — APPROVED** [review]
Yep, everything builds for me with this.

Looking forward to all the helpful `lib.warn`s we will be able to add in two weeks.

**Comment by amjoseph-nixpkgs** [comment]
Ugh, unfortunately my workaround for the fact that `pythonPackages` isn't spliced no longer works with this.  So I don't know how to make `pyqt5` cross-compile anymore.

**Comment by Artturin** [comment]
python3Packages is spliced.

File: pkgs/top-level/qt5-packages.nix [hunk 1]
```diff
@@ -7,13 +7,19 @@
...
+, __splicedPackages
+, makeScopeWithSplicing'
+, generateSplicesForMkScope
 }:
 
-(lib.makeScope pkgs.newScope ( self:
-
 let
+  pkgs = __splicedPackages;
+  qt5 = __splicedPackages.qt5;
```
Comment (Artturin) [inline pkgs/top-level/qt5-packages.nix:17]: @amjoseph-nixpkgs 
this is the issue

```
nix-repl> pkgsCross.aarch64-multiplatform.pkgsBuildTarget.targetPackages.libsForQt5.qtbase ? __spliced
true
```


Fixing it..

Comment (Artturin) [reply pkgs/top-level/qt5-packages.nix:17]: https://github.com/NixOS/nixpkgs/pull/267323

**Pull request merged**